### PR TITLE
[5556] Studio Search and Search Dialogs have Path filter that limits to current path only

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5556

### CODE REVIEW

File: ui/app/src/components/SiteSearchPathSelector/PathSelector.tsx [hunk 1]
```diff
@@ -138,28 +182,53 @@ export function PathSelector(props: PathSelectorProps) {
...
+          startAdornment={<SearchIcon className={classes.pathSelectorSearchIcon} />}
+          endAdornment={
+            !disabled && value ? (
+              <IconButton onClick={onClean} size="small">
+                <CloseIcon />
+              </IconButton>
+            ) : null
+          }
+        />
+      </Paper>
```
Comment (rart) [inline ui/app/src/components/SiteSearchPathSelector/PathSelector.tsx]: Input should only shown when a base path is selected

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/SiteSearchPathSelector/PathSelector.tsx [hunk 2]
```diff
@@ -66,16 +71,54 @@ const useStyles = makeStyles()((theme) => ({
...
+    id: 'words.assets',
+    defaultMessage: 'Assets'
+  },
+  templates: {
+    id: 'words.templates',
+    defaultMessage: 'Templates'
+  },
+  scripts: {
+    id: 'words.scripts',
+    defaultMessage: 'scripts'
```
Comment (rart) [inline ui/app/src/components/SiteSearchPathSelector/PathSelector.tsx]:
```suggestion
    defaultMessage: 'Scripts'
```

File: ui/app/src/components/SiteSearchPathSelector/PathSelector.tsx [hunk 3]
```diff
@@ -138,28 +182,55 @@ export function PathSelector(props: PathSelectorProps) {
...
-              <CloseIcon />
-            </IconButton>
-          ) : null
-        }
-      />
-    </Paper>
+    <>
+      <FormControl className={classes.basePathSelectorContainer}>
+        <RadioGroup value={radioBasePath} onChange={handleBasePathChange}>
+          <FormControlLabel value="" control={<Radio size="small" />} label="Any path" />
```
Comment (rart) [inline ui/app/src/components/SiteSearchPathSelector/PathSelector.tsx]: Label lacking i18n

File: ui/app/src/components/SiteSearchPathSelector/PathSelector.tsx [hunk 4]
```diff
@@ -138,28 +182,55 @@ export function PathSelector(props: PathSelectorProps) {
...
-            <IconButton onClick={onClean} size="small">
-              <CloseIcon />
-            </IconButton>
-          ) : null
-        }
-      />
-    </Paper>
+    <>
+      <FormControl className={classes.basePathSelectorContainer}>
+        <RadioGroup value={radioBasePath} onChange={handleBasePathChange}>
```
Comment (rart) [inline ui/app/src/components/SiteSearchPathSelector/PathSelector.tsx:196]: The value of the base path radio should still be checked when a subpath of his is selected. This won't work.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/SiteSearchPathSelector/PathSelector.tsx (hunk 4 above)
Comment (jvega190) [reply ui/app/src/components/SiteSearchPathSelector/PathSelector.tsx:196]: it does work, the radioBasePath updates to be the `root` of the path so it matches the radio values, so they're properly checked. (look at comment tag)

File: ui/app/src/components/SiteSearchPathSelector/PathSelector.tsx [hunk 5]
```diff
@@ -90,6 +137,7 @@ export function PathSelector(props: PathSelectorProps) {
   const [path, setPath] = useState<string>(value ?? '');
   const idSuccess = 'pathSelectionSuccess';
   const idCancel = 'pathSelectionCancel';
+  const radioBasePath = path.split('/')[1] ? `/${path.split('/')[1]}` : '';
```
Comment (jvega190) [inline ui/app/src/components/SiteSearchPathSelector/PathSelector.tsx:140]: @rart here

**Review by rart — APPROVED** [review]

**Pull request merged**